src/tests/test_survey.py

```diff
@@ -0,0 +1,148 @@
+import os
+import sys
+import json
+import pytest
+
+parent_path = os.path.dirname(os.getcwd())
+sys.path.append(parent_path)
+
+from src.config import TestConfig  # NOQA
+from src.surfing_penguin.routes import blueprint  # NOQA
+from src.surfing_penguin import create_app  # NOQA
+from src.surfing_penguin.models import Survey, Question # NOQA
+import src.surfing_penguin.db_interface.SurveyFunctions as SurveyFunctions  # NOQA
+
+
+@pytest.fixture(scope='class')
+def app():
+    app = create_app(TestConfig)
+    app.register_blueprint(blueprint)
+    # TODO: check if using the correct database
+    return app
+
+
+@pytest.fixture(scope='class')
+def session(app):
+    from src.surfing_penguin.extensions import session
+    return session
+
+
+@pytest.fixture
+def client(app):
+    test_client = app.test_client()
+
+    def teardown():
+        pass
+        # Databases and resourses have to be freed at
+        # the end.
+
+    # request.addfinalizer(teardown)
+    return test_client
+
+
+@pytest.fixture
+def post_json(client, url, json_dict):
+    """Send dictionary json_dict as a json to the specified url """
+    return client.post(url, data=json.dumps(json_dict),
+                       content_type='application/json')
+
+
+def json_of_response(response):
+    """Decode json from response"""
+    return json.loads(response.data.decode('utf8'))
+
+
+# TODO: separate test functions to different classes
+@pytest.fixture
+def survey():
+    new_survey = Survey("test")
+    return new_survey
+
+
+# data for questions in survey
+@pytest.fixture
+def question_data():
+    data = [
+        {
+            'idx': 1,
+            'title': "testQ1",
+            'content': "Q1content"
+        },
+        {
+            'idx': 2,
+            'title': "testQ2",
+            'content': "Q2content"
+            }
+    ]
+    return data
+
+
+@pytest.mark.usefixtures('session')
+class TestSurvey():
+    # Warning: Test functions in this class
+    # use the same database, so be mindful of the
+    # sequece to call them.
+
+    """ testing models """
+    def test_Survey_model(self, survey):
```

Comment:
Do not mix upper case with lower case for function naming. Use lower case with underscores for functions. See https://google.github.io/styleguide/pyguide.html#Naming